### DISCUSSION TITLE
NSFW Filter

### DISCUSSION THREAD

**Discussion by Nomidos** [zanllp/infinite-image-browsing#511] [Ideas]
This a great addition to A1111, I'd even say it's essential to anyone who uses it. I had an idea about privacy that I wanted to run by you. Could you please consider adding a NSFW filter that allows you to conceal NFSW images from the regular browsing window? Something that doesn't make it obvious that it's NFSW content but hides the images maybe even in a separate browser window.  Cheers!

**Comment by Woisek** [comment]
Simplest way: Move all NSFW into its own folder, don't browse the folder.
Other: Tag all NSFW and then let show all except NSFW images (if that's currently possible in that way)

**Comment by zanllp** [comment]
How did this discussion get moved to the top? I seem to remember seeing this discussion but then forgot about it. 


I think a more elegant solution would be to add your blocklist words or tags in the global settings page. I will blur or block the images during loading based on the detailed generation of the image, just like mentioned in issue #279. At that time, I wanted to do it, but there were many higher priority tasks waiting for me. After waiting for a long time, I became unsure if it was still needed. I'm worried that no one will use it after it's completed.
